### ISSUE TITLE
More Requests !

### ISSUE THREAD

**Issue by GoogleCodeExporter** [chrsmithdemos/reaver-wps#176]
```
Me again ! :o)

I am sorry about all these requests but Reaver is such a fantastic tool !!

When Reaver detects "rate limiting detected" would it be a good idea for Reaver 
to automatically change the MAC address and try again ?  Just in case blocking 
is MAC based.

Another idea is when Reaver detects rate limiting could it time exactly how 
long the time out period is so when it happens again Reaver knows how long to 
wait (perhaps add a second or two) before starting again ?  This way users know 
Reaver is optimised to the AP being tested.

Thanks.
```

Original issue reported on code.google.com by `keyfo...@veryrealemail.com` on 24 Jan 2012 at 5:44

**Comment by GoogleCodeExporter** [comment]
```
I have never seen a MAC-based block, blocks are always global. If anyone finds 
an implementation that actually does only block a single MAC, then maybe I'll 
add it.

I don't think there's much advantage to Reaver timing the lock out period. 
Reaver will check every X seconds (default is 60) to see if WPS has been 
unlocked, and as soon as it is unlocked Reaver will continue If you want Reaver 
to check more often, you can set the period with --lock-delay option (i.e., 
'--lock-delay=1' will cause Reaver to re-check the WPS lock status every 1 
second).
```

Original comment by `cheff...@tacnetsol.com` on 24 Jan 2012 at 10:26
- Changed state: **WontFix**

**Comment by GoogleCodeExporter** [comment]
```
Ok fair enough about the MAC thing, I just assumed thats how it was done, but I 
am certain you know considerably more about it than I do !  :o)

I understand about the time out option you have given us in Reaver and I have 
used it today which is what prompted my request.  I just thought it would be a 
neat trick to tune Reaver automatically to each AP's time outs rather than 
waiting longer than necessary.  It was just an optimising fine tune sort of 
thing that probably wouldn't gain the user much time in the end anyway.

I am however still very keen on the option to count down ,tested another today 
that started with a 9 !! Ha ha !

Thanks.
```

Original comment by `keyfo...@veryrealemail.com` on 24 Jan 2012 at 11:04
